app/views/messages/_message.html.erb

```diff
@@ -162,37 +162,37 @@
             </menu>
           </div>
         <% end %>
-
-        <% if message.has_document_image? %>
-          <dialog class="modal no-focus-outline" data-modal-target="dialog" data-role="image-modal">
-            <main class="modal-box max-w-5xl p-0 rounded-none">
-              <article
-                data-controller="image-loader"
-                data-image-loader-url-value="<%= message.document_image_path(:large) %>"
-                data-turbo-permanent
-                class="flex flex-col md:flex-row justify-center"
-              >
-                <%= image_tag message.document_image_path(:large, fallback: ""),
-                  data: {
-                    image_loader_target: "image",
-                    action: %|
-                      error->image-loader#retryAfterDelay
-                      load->image-loader#show
-                    |
-                  },
-                  class: "w-full h-auto"
-                %>
-                <div data-role="image-loader" class="hidden p-5" data-image-loader-target="loader">
-                  <%= spinner size: 6, class: "text-black dark:text-white" %>
-                </div>
-              </article>
-            </main>
-            <form method="dialog" class="modal-backdrop no-focus-outline">
-              <button id="modal-backdrop">close</button>
-            </form>
-          </dialog>
-        <% end %>
       </div>
+
+      <% if message.has_document_image? %>
+        <dialog class="modal no-focus-outline" data-modal-target="dialog" data-role="image-modal">
+          <main class="modal-box max-w-5xl p-0 rounded-none">
+            <article
+              data-controller="image-loader"
+              data-image-loader-url-value="<%= message.document_image_path(:large) %>"
+              data-turbo-permanent
+              class="flex flex-col md:flex-row justify-center"
+            >
+              <%= image_tag message.document_image_path(:large, fallback: ""),
+                data: {
+                  image_loader_target: "image",
+                  action: %|
+                    error->image-loader#retryAfterDelay
+                    load->image-loader#show
+                  |
+                },
+                class: "w-full h-auto"
+              %>
+              <div data-role="image-loader" class="hidden p-5" data-image-loader-target="loader">
+                <%= spinner size: 6, class: "text-black dark:text-white" %>
+              </div>
+            </article>
+          </main>
+          <form method="dialog" class="modal-backdrop no-focus-outline">
+            <button id="modal-backdrop">close</button>
+          </form>
+        </dialog>
+      <% end %>
```

Comment:
The modal was inside of the message footer div which shows & hides on hover. I think that's why it sometimes was failing the `modal.visible?` check